aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsEndpointDecorator.kt

```diff
@@ -109,24 +109,26 @@ class EndpointConfigCustomization(
         "aws_types" to awsTypes(runtimeConfig).asType(),
     )
 
-    override fun section(section: ServiceConfig): Writable =
-        writable {
-            when (section) {
-                is ServiceConfig.ConfigStruct -> rustTemplate(
-                    "pub (crate) endpoint_resolver: std::sync::Arc<dyn #{SmithyResolver}<#{PlaceholderParams}>>,",
-                    *codegenScope,
-                )
-
-                is ServiceConfig.ConfigImpl -> emptySection
-                is ServiceConfig.BuilderStruct ->
-                    rustTemplate(
-                        "endpoint_resolver: Option<std::sync::Arc<dyn #{SmithyResolver}<#{PlaceholderParams}>>>,",
-                        *codegenScope,
-                    )
-
-                ServiceConfig.BuilderImpl ->
-                    rustTemplate(
-                        """
+    override fun section(section: ServiceConfig): Writable = writable {
+        when (section) {
+            is ServiceConfig.ConfigStruct -> rustTemplate(
+                "pub (crate) endpoint_resolver: std::sync::Arc<dyn #{SmithyResolver}<#{PlaceholderParams}>>,",
+                *codegenScope,
+            )
+            is ServiceConfig.ConfigImpl -> rustTemplate(
```

Comment:
For this error in CI:
>  error: Unapproved external type `aws_endpoint::Params` referenced in public API

I think we should comment this getter out for now since we definitely don't want the placeholder params to become part of the API. We can add a TODO comment that links to an issue to add this once our endpoints 2.0 project is completed.